tests/quantization/higgs/test_higgs.py

```diff
@@ -0,0 +1,254 @@
+# coding=utf-8
+# Copyright 2024 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import gc
+import tempfile
+import unittest
+
+from transformers import AutoConfig, AutoModelForCausalLM, AutoTokenizer, HiggsConfig, OPTForCausalLM
+from transformers.testing_utils import (
+    require_accelerate,
+    require_flute_hadamard,
+    require_torch_gpu,
+    require_torch_multi_gpu,
+    slow,
+    torch_device,
+)
+from transformers.utils import is_accelerate_available, is_torch_available
+
+
+if is_torch_available():
+    import torch
+
+if is_accelerate_available():
+    from accelerate import init_empty_weights
+
+
+# @require_torch_gpu
+# class HiggsConfigTest(unittest.TestCase):
+#     def test_to_dict(self):
+#         """
+#         Simple test that checks if one uses a config and converts it to a dict, the dict is the same as the config object
+#         """
+#         quantization_config = HiggsConfig()
+#         config_to_dict = quantization_config.to_dict()
+
+#         for key in config_to_dict:
+#             self.assertEqual(getattr(quantization_config, key), config_to_dict[key])
+
+#     def test_from_dict(self):
+#         """
+#         Simple test that checks if one uses a dict and converts it to a config object, the config object is the same as the dict
+#         """
+#         dict = {"linear_weights_not_to_quantize": ["embed_tokens.weight", "lm_head.weight"], "quant_method": "higgs"}
+#         quantization_config = HiggsConfig.from_dict(dict)
+
+#         self.assertEqual(dict["linear_weights_not_to_quantize"], quantization_config.linear_weights_not_to_quantize)
+#         self.assertEqual(dict["quant_method"], quantization_config.quant_method)
+
+
+@slow
+@require_torch_gpu
+@require_flute_hadamard
+@require_accelerate
+# @require_read_token
+class HiggsTest(unittest.TestCase):
+    model_name = "meta-llama/Meta-Llama-3.1-8B"
```

Comment:
can we use a smaller model like a tiny llama ? This will be better for our CI thanks !

Reply:
Sadly, no. FLUTE is only compiled for specific matrix shapes, for now.
TinyLlama is not among those shape. Nor is any model smaller than 8B.

Reply:
sounds good ! Do we have a check somewhere for the matrix shape then ? If a run this on a tinyllama, what kind of error will I get. If it is possible to detect the issue beforehand and output a nice error message, that would be nice.